### PR TITLE
fix proof hash

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Get sha3 hash from proof before signing.

See https://github.com/soramitsu/soranet-eth/blob/develop/notary-eth-integration-test/src/main/kotlin/jp/co/soramitsu/soranet/eth/integration/helper/ContractTestHelper.kt#L199

### CODE REVIEW

File: eth-bridge/src/main/kotlin/jp/co/soramitsu/soranet/eth/bridge/ReferendumHashProofHandler.kt [hunk 1]
```diff
@@ -91,7 +92,8 @@ class ReferendumHashProofHandler(
             null
     }
 
-    private fun createProof(hash: String): String {
+    private fun createProof(proof: String): String {
+        val hash = hashToProve(proof)
```
Comment (pgolovkin) [inline eth-bridge/src/main/kotlin/jp/co/soramitsu/soranet/eth/bridge/ReferendumHashProofHandler.kt]: Looks good. 
However this method may be refactored. We have the same block of code at line 139.
We can avoid such error in future with it.

Comment (Alexey-N-Chernyshov) [reply eth-bridge/src/main/kotlin/jp/co/soramitsu/soranet/eth/bridge/ReferendumHashProofHandler.kt]: Where is 139 line? 👀 
There is 119 lines total.
btw I updated the fix.

**Review by pgolovkin — COMMENTED** [review]
Looks good, please, review my comments.

File: eth-bridge/src/main/kotlin/jp/co/soramitsu/soranet/eth/bridge/ReferendumHashProofHandler.kt (hunk 1 above)
Comment (pgolovkin) [reply eth-bridge/src/main/kotlin/jp/co/soramitsu/soranet/eth/bridge/ReferendumHashProofHandler.kt]: Check line 139 in jp.co.soramitsu.soranet.eth.bridge.WithdrawalProofHandler#createProof

**Review by Mingela — APPROVED** [review]

**Review by pgolovkin — APPROVED** [review]

**Pull request merged**